ir/global.go

```diff
@@ -35,6 +88,9 @@ type Global struct {
 	Init constant.Constant
 	// Immutability of the global variable.
 	IsConst bool
+	// Linkage type of the global variable.
+	LinkageType LinkageType
```

Comment:
To be consistent with the naming of `UnnamedAddr UnnamedAddrType`, consider the following change:
```diff
-LinkageType LinkageType
+Linkage LinkageType
```

Reply:
I did consider that. I went with LinkageType for the field name because that's how the LLVM documentation refers to it: linkage types. The type could've been called `LinkageTypeType`, but that's obviously silly.

Reply:
Haha, yes, LinkageTypeType is silly indeed. Ok, good to see you gave it some thought. I've fine with `LinkageType` if you prefer.